Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -91,4 +179,151 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::symbol_species_getter)
     return vm.this_value();
 }
 
+// get RegExp.input, https://github.com/tc39/proposal-regexp-legacy-features#get-regexpinput
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_getter)
+{
+    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();
+
+    // 1. Return ? GetLegacyRegExpStaticProperty(%RegExp%, this value, [[RegExpInput]]).
+    return js_string(vm, regexp_constructor->legacy_static_properties().input());
+}
+
+// get RegExp.$_, https://github.com/tc39/proposal-regexp-legacy-features#get-regexp_
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_alias_getter)
+{
+    // Keep the same implementation with `get RegExp.input`
+    return input_getter(vm);
+}
+
+// SetLegacyRegExpStaticProperty( C, thisValue, internalSlotName, val ) https://github.com/tc39/proposal-regexp-legacy-features#setlegacyregexpstaticproperty-c-thisvalue-internalslotname-val-
+static void set_legacy_regexp_static_property(VM& vm, Function<void(Utf16View const&)>& property_setter, String const& new_value)
+{
+    (void)vm;
+
+    // FIXME: 1. Assert C is an object that has an internal slot named internalSlotName.
+    // FIXME: 2. If SameValue(C, thisValue) is false, throw a TypeError exception.
+
+    // 3. Let strVal be ? ToString(val).
+    auto str_value = Utf16String(new_value).view();
+
+    // 4. Set the value of the internal slot of C named internalSlotName to strVal.
+    property_setter(str_value);
+}
+
+// set RegExp.input, https://github.com/tc39/proposal-regexp-legacy-features#set-regexpinput--val
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_setter)
+{
+    auto new_value = TRY(vm.argument(0).to_string(vm));
+    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();
+    Function<void(Utf16View const&)> property_setter = [&](Utf16View const& new_value) {
+        regexp_constructor->legacy_static_properties().set_input(new_value);
+    };
+
+    set_legacy_regexp_static_property(vm, property_setter, new_value);
+
+    // Return the new value.
+    return js_string(vm, new_value);
```

Comment:
...and with all of the above, this can become:

```suggestion
    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();
    auto property_setter = [&](String new_value) {
        regexp_constructor->legacy_static_properties().set_input(move(new_value));
    };

    // 1. Perform ? SetLegacyRegExpStaticProperty(%RegExp%, this value, [[RegExpInput]], val).
    TRY(set_legacy_regexp_static_property(vm, *regexp_constructor, vm.this_value(), property_setter, vm.argument(0)));
```